Changes the link references to "master" instead of a specific old branch

sterlingbaldwin opened the pull request:
I went to go add the F to C conversion in the weather story provider only to find it already existed! Turns out the link to the weather provider in the README went to an old version of the code, so I've updated the links to point at master instead. If there's some reason to point at the old version, feel free to just close this, but having it point at master might help other people not be confused.

j6k4m8 commented:
[sigh] if only we could all just use celsius finally... 🙄 

Thank you so much for contributing, sorry for the confusion! The original reason the links point to a specific commit was to make it so that changes to the code (and therefore line-numbers) in the file didn't break where the links pointed. But that'll all get sorted out eventually when we move StoryProviders each to their own file, and I agree this makes a lot more sense in the meantime. Good to merge? :D

sterlingbaldwin commented:
This is a cool project BTW, I just got my reMarkable 2 and immediately started hunting around for something like this.

j6k4m8 commented:
Nice! I was so surprised to see that nothing like this existed too!

Let me know if it works for you or if you encounter any silliness!!

sterlingbaldwin commented:
I started having a bit too much fun with this, and ended up making some fairly comprehensive changes. I added a couple things to my fork:
  * I thought the articles were a bit short, so for the rss feed I added a feature to pull down the full text of the articles and extract just the bits that you actually want to read.
  * To do this, I had to add per-site parsers, right now I just have a generic one, one for the nyt, npr, and arstechnica.
  * I added an epub output option. This is _much_ faster then writing it out as a pdf, and I was hoping to be able to use the table of contents to jump directly to stories, but alas the build in remarkable epub reader is, shall we say, feature poor.
  * The biggest change is that I broke the whole thing apart into a bunch of different files. 90% of the code is the same, but now theres a file for each of the story providers.
  * I changed the setup.py scripts install to use a __main__ function instead of the cli function. 

Not strictly code related, but I also setup an n8n workflow that generates and uploads the paper to me every morning automatically. Its pretty fun!

j6k4m8 commented:
This is awesome @sterlingbaldwin!! Would you be willing to PR in these changes? How atomic are they (i.e. possible to do one PR per bullet, or is it all together)? Would love to upstream these, if you're interested!!! :D
